ToolChain/Debugger/Debugger.py

```diff
@@ -589,8 +627,24 @@ def addBreakPoint(self, address=None):
                 self.debugLog("Error while processing address or symbol {}".format(address,))
                 return
 
-        self.breakPoints.append(address)
+        if address in self.breakPoints:
+            self.debugLog("Error breakpoint already exist")
+            return
+        else:
+            self.breakPoints.append(address)
+
         self.breakPoints.sort()
+        self.writeBreakPointFile()
+
+    def writeBreakPointFile(self):
+        """
+        This function writes the breakpoints stored in self.breakpoints to a file
+        :return:
+        """
+        file = open((self.inputFile.split(".")[0]+".bp"), "w")
```

Comment:
When the path is validated, the "bp" file extension is not enforced. (Line 118, Debugger.py)
The same logic should be used here otherwise we expose ourselves to a bug.

Reply:
Actually this is currently causing a bug in the case the user specify a different BP file name than input name. See following trace:
```
~/Desktop/spartacus$ python3.6 Debugger.py -i testFiles/LoneBall/LoneBall.bin -bp test.bp
Debug session about to begin, following options will be used
  input file:             testFiles/LoneBall/LoneBall.bin
  symbols file:             testFiles/LoneBall/LoneBall.sym
Building Capua execution environment
Loading ('testFiles/LoneBall/LoneBall.bin',) in memory
Done loading ('testFiles/LoneBall/LoneBall.bin',) into memory
Loading symbols from file testFiles/LoneBall/LoneBall.sym
Done loading symbols
Loading breakpoints from file test.bp
Done loading breakpoints
Debugging session is ready to be used. Have fun!

Next instruction to be executed:
0x40000000 : MOV #0x400004fb $A 
('testFiles/LoneBall/LoneBall.bin',): db
db

Breakpoints list:

Next instruction to be executed:
0x40000000 : MOV #0x400004fb $A 
('testFiles/LoneBall/LoneBall.bin',): ss
ss
Loaded symbols are:
0x40000907 : display.DCC_LOOP
0x400008de : display.DC_LOOP
0x400008ae : display.DISPLAY_CHAR
0x400008cc : display.DISPLAY_CLEAR
0x400008f9 : display.DISPLAY_CLEAR_COL
0x4000086d : display.DISPLAY_LINE
0x40000883 : display.DL_PRINT_LOOP

Next instruction to be executed:
0x40000000 : MOV #0x400004fb $A 
('testFiles/LoneBall/LoneBall.bin',): b display.DISPLAY_CHAR
b display.DISPLAY_CHAR

Next instruction to be executed:
0x40000000 : MOV #0x400004fb $A 
('testFiles/LoneBall/LoneBall.bin',): db
db

Breakpoints list:
 0 - 0x400008ae -> display.DISPLAY_CHAR

Next instruction to be executed:
0x40000000 : MOV #0x400004fb $A 
('testFiles/LoneBall/LoneBall.bin',): quit
quit
~/Desktop/spartacus$ ls
Assembler.py      Debugger.py          HD.bin      README.md  ToolChain
CapuaEnvironment  Documentation        LICENSE.md  test.bp
Configuration     HardDriveCreator.py  Linker.py   testFiles
~/Desktop/spartacus$ cat test.bp 
~/Desktop/spartacus$ 
```